echopype/clean/api.py

```diff
@@ -74,3 +85,206 @@ def remove_noise(ds_Sv, ping_num, range_sample_num, noise_max=None, SNR_threshol
+    )
+    return return_mask
+
+
+def get_impulse_noise_mask(
+    source_Sv: xr.Dataset,
+    desired_channel: str,
+    thr: Union[Tuple[float, float], int, float],
+    m: Optional[Union[int, float]] = None,
+    n: Optional[Union[int, Tuple[int, int]]] = None,
+    erode: Optional[List[Tuple[int, int]]] = None,
+    dilate: Optional[List[Tuple[int, int]]] = None,
+    median: Optional[List[Tuple[int, int]]] = None,
+    method: str = "ryan",
+) -> xr.DataArray:
+    """
+    Algorithms for masking Impulse noise.
+
+    Parameters
+    ----------
+    source_Sv: xr.Dataset
+        Dataset  containing the Sv data to create a mask
+    desired_channel: str
+        Name of the desired frequency channel.
+    thr: float or tuple
+        User-defined threshold value (dB) (ryan and ryan iterable) o
+        r a 2-element tuple specifying the range of threshold values (wang).
+    m: int or float, optional
+        Vertical binning length (in number of samples or range) (ryan and ryan iterable).
+        Defaults to None.
+    n: int or tuple, optional
+        Number of pings either side for comparisons (ryan),
+        or a 2-element tuple specifying the range (ryan iterable).
+        Defaults to None.
+    erode: List of 2-element tuples, optional
+        List indicating the window's size for each erosion cycle (wang). Defaults to None.
+    dilate: List of 2-element tuples, optional
+        List indicating the window's size for each dilation cycle (wang). Defaults to None.
+    median: List of 2-element tuples, optional
+        List indicating the window's size for each median filter cycle (wang). Defaults to None.
+    method: str, optional
+        The method (ryan, ryan iterable or wang) used to mask impulse noise. Defaults to 'ryan'.
+
+    Returns
+    -------
+    xr.DataArray
+        A DataArray consisting of a mask for the Sv data, wherein True values signify
+        samples that are free of noise.
+    """
+
+    # Our goal is to have a mask where True represents samples that are NOT impulse noise.
+    # So, we negate the obtained mask.
+
+    if method == "ryan":
+        impulse_mask_ryan = _find_impulse_mask_ryan(source_Sv, desired_channel, m, n, thr)
+        noise_free_mask = ~impulse_mask_ryan
+    elif method == "ryan_iterable":
+        impulse_mask_ryan_iterable = _find_impulse_mask_ryan_iterable(
+            source_Sv, desired_channel, m, n, thr
+        )
+        noise_free_mask = ~impulse_mask_ryan_iterable
+    elif method == "wang":
+        impulse_mask_wang = _find_impulse_mask_wang(
+            source_Sv, desired_channel, thr, erode, dilate, median
+        )
+        noise_free_mask = ~impulse_mask_wang
+    else:
+        raise ValueError(f"Unsupported method: {method}")
+
+    return noise_free_mask
+
+
+def get_attenuation_mask(
+    source_Sv: Union[xr.Dataset, str, pathlib.Path],
+    desired_channel: str,
+    mask_type: str = "ryan",
+    **kwargs,
```

Comment:
Instead of `kwargs`, suggest using the same approach as in [this comment](https://github.com/OSOceanAcoustics/echopype/pull/1142/files#r1334749088) and provide explanation and parameter requirements in the docstring.